webui/react/src/components/LogViewerCore.tsx

```diff
@@ -188,7 +188,8 @@ const LogViewerCore: React.FC<Props> = ({
     if (!local.current.isScrollReady) return;
 
     local.current.isOnTop = visibleStartIndex === 0;
-    local.current.isOnBottom = visibleStopIndex === logs.length - 1;
+    // mismatch between visibleStopIndex and logs updates causes flickering and disappearing logs:
+    local.current.isOnBottom = visibleStopIndex + 1 === logs.length - 1;
```

Comment:
**question:** I did some console logging around here:
```
console.log('isOnBottom', local.current.isOnBottom);
console.log('start', visibleStartIndex, 'stop', visibleStopIndex, 'logs.length', logs.length);
```
And it looks like majority of the time, `isOnBottom` is false even on the tail end because `visibleStopIndex` is usually equal to `logs.length - 1`. So maybe it would be better to use `>=` instead of `===` for comparison?

Reply:
![Screen Shot 2022-02-15 at 11 22 12 AM](https://user-images.githubusercontent.com/220971/154124752-705b909d-27df-424e-9a41-5a071b4c87c3.png)

Here's the example of where the tailing is enabled button doesn't show because `visibleStopIndex` is exactly `logs.length - 1`